### PR TITLE
Fixed the TC PXBackupEndToEndBackupAndRestoreWithUpgrade

### PR DESCRIPTION
**Pull request by sabrarhussaini**
Fixed the TC **PXBackupEndToEndBackupAndRestoreWithUpgrade** to work with Non-Px clusters.

**To accommodate the changes, the below two helper methods were added:**

- [x] WatchAndStopCloudsnapBackup
- [x] IsPxInstalled

**What this PR does / why we need it**:
The initial PR merged for this TC didn't consider the Non-Px clusters.

**Which issue(s) this PR fixes** (optional)
Closes #PB-7587

**Special notes for your reviewer**:
Px cluster: (Vanilla)
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-Upgrade/job/px-backup-upgrade-test-vanilla/93/
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-Upgrade/job/px-backup-upgrade-test-vanilla/96/

Non-Px cluster: (IKS)
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-with-Stork-Upgrade/job/Px-Backup+Stork-Upgrade-iks-non-px-csi-offload/35/
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/upgrade-tests/job/Px-Backup-with-Stork-Upgrade/job/Px-Backup+Stork-Upgrade-iks-non-px-csi-offload/36/

### CODE REVIEW

File: tests/backup_helper.go [hunk 1]
```diff
@@ -10368,3 +10441,12 @@ func UpdateDriverWithEnvVariable(envVariables map[string]string) error {
...
 	return nil
 }
+
+// IsPxInstalled checks if PX is installed by verifying the length of the list returned by GetStorageDriverNodes.
+func IsPxInstalled() bool {
+	numOfNodes := len(node.GetStorageDriverNodes())
+	if numOfNodes > 0 {
+		return true
+	}
+	return false
```
Comment (mkoppal-px) [inline tests/backup_helper.go:10451]: `return numOfNodes > 0`

File: tests/backup/backup_upgrade_test.go [hunk 2]
```diff
@@ -655,6 +585,72 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", Label(TestCase
...
+						err := WatchAndStopCloudsnapBackup(pvc.Name, pvc.Namespace, 2*time.Hour, ctx)
+						if err != nil {
+							dash.VerifySafely(err, nil, fmt.Sprintf("watching and stopping cloudsnap backup for PVC [%s] in namespace [%s]: %v", pvc.Name, pvc.Namespace, err))
+						} else {
+							log.InfoD("Cloudsnap backup watch and stop process completed successfully for PVC [%s] in namespace [%s].", pvc.Name, pvc.Namespace)
+						}
+					}(pvc)
+				}
+			}
+		})
```
Comment (mkoppal-px) [inline tests/backup/backup_upgrade_test.go:621]: Add an else block and mention that we are skipping this step since it is a non-px cluster

**Review by mkoppal-px — APPROVED** [review]
LGTM except for a couple of minor comments

File: tests/backup_helper.go [hunk 3]
```diff
@@ -9770,6 +9770,79 @@ func GetCloudsnapStatus() (map[string]CloudsnapStatus, error) {
...
+		case <-ticker.C:
+			statusMap, err := getCloudsnapStatus()
+			if err != nil {
+				return fmt.Errorf("failed to get cloudsnap status from cluster: %w", err)
+			}
+			found := false
+			for key, value := range statusMap {
+				if strings.Contains(key, fmt.Sprintf("%s-%s", namespace, pvcName)) {
+					found = true
+					// log.Infof("Key - %s", key)
```
Comment (kshithijiyer-px) [inline tests/backup_helper.go:9807]: Can we remove these lines if not needed?

**Review by kshithijiyer-px — APPROVED** [review]
Minor comment other than that looks good to me.

File: tests/backup_helper.go (hunk 1 above)
Comment (sabrarhussaini) [reply tests/backup_helper.go]: Addressed

File: tests/backup/backup_upgrade_test.go (hunk 2 above)
Comment (sabrarhussaini) [reply tests/backup/backup_upgrade_test.go:629]: Addressed

**Pull request merged**